lib/gitlab/client/branches.rb

```diff
@@ -47,7 +47,7 @@ def branch(project, branch)
     # @option options [Boolean] :developers_can_merge True to allow developers to merge into the branch (default = false)
     # @return [Gitlab::ObjectifiedHash] Details about the branch
     def protect_branch(project, branch, options = {})
-      post("/projects/#{url_encode project}/protected_branches", body: {name: branch}.merge(options))
+      put("/projects/#{url_encode project}/repository/branches/#{url_encode branch}/protect", query: options)
```

Comment:
Docs are saying that this API endpoint is deprecated in favor of `POST /projects/:id/protected_branches`. https://docs.gitlab.com/ce/api/branches.html#protect-repository-branch